### DISCUSSION TITLE
RailRoad Diagram

### DISCUSSION THREAD

**Discussion by aarmn** [pest-parser/pest#807] [Ideas]
Is it possible to automatically generate railroad diagram based on pest files? Or is railroad a subset of pest features and therefore all pest grammars cannot be mapped to a railroad diagram?

**Comment by tomtau** [comment]
https://github.com/pest-parser/pest/discussions/803

It sounds cool, but it seems it hasn't been explored yet.

**Comment by nu11ptr** [comment]
I just wrote this for myself in case it works for you: https://crates.io/crates/pest_railroad_gen
